src/libs/sys-util-rs/src/device.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2019-2021 Alibaba Cloud
+// Copyright (c) 2019-2021 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::fs;
+use std::io::Result;
+use std::path::PathBuf;
+
+use nix::sys::stat::{self, SFlag};
+
+use crate::{eother, sl};
+
+const SYS_DEV_BLOCK_PATH: &str = "/sys/dev/block";
+
+/// Get major and minor number of the device or of the device hosting the regular file/directory.
+pub fn get_devid(file: &str) -> Result<(u64, u64)> {
+    let fstat =
+        stat::stat(file).map_err(|e| eother!("get_devid: failed to stat {}: {:?}", file, e))?;
+
+    match SFlag::from_bits_truncate(fstat.st_mode) {
+        // For block device, check major/minor of the device special file itself
```

Comment:
Handling these two very different cases with the same function sounds like a really bad idea to me.  I have trouble imagining a caller that wants this, which isn't buggy.

Reply:
Haha, you missed one usage case: blkio cgroup.
I will rename this function to get_devid_for_blkio_cgroup().

Reply:
Rename the function to "get_devid_for_blkio_cgroup()".